tests/checks/command-not-found.fish

```diff
@@ -2,17 +2,16 @@
 set -g PATH
 $fish -c "nonexistent-command-1234 banana rama"
 #CHECKERR: fish: Unknown command: nonexistent-command-1234
-#CHECKERR: fish: 
 #CHECKERR: nonexistent-command-1234 banana rama
 #CHECKERR: ^~~~~~~~~~~~~~~~~~~~~~~^
-$fish -C 'function fish_command_not_found; echo cmd-not-found; end' -ic "nonexistent-command-1234 1 2 3 4"
-#CHECKERR: cmd-not-found
-#CHECKERR: fish: 
+$fish -C 'function fish_command_not_found; echo cmd-not-found; end' -c "nonexistent-command-1234 1 2 3 4"
+##CHECKERR: cmd-not-found
+#CHECKERR: fish: Unknown command: nonexistent-command-1234
 #CHECKERR: nonexistent-command-1234 1 2 3 4
 #CHECKERR: ^~~~~~~~~~~~~~~~~~~~~~~^
 $fish -C 'function fish_command_not_found; echo command-not-found $argv; end' -c "nonexistent-command-abcd foo bar baz"
-#CHECKERR: command-not-found nonexistent-command-abcd foo bar baz
-#CHECKERR: fish: 
+##CHECKERR: command-not-found nonexistent-command-abcd foo bar baz
+#CHECKERR: fish: Unknown command: nonexistent-command-abcd
```

Comment:
this is a regression, unfortunately. need to figure out why this particular needed case fails.

Reply:
This fails because this runs a fish - `$fish` above - that isn't interactive.

So because you only made it execute `fish_command_not_found` if fish is interactive, and this tries to override fish_command_not_found, it no longer works.

I'm not sure if that's okay.

Reply:
Yes, I'm not 100% that this doesn't need more tweaks, but I felt it was a start. A bunch of the error responses changed because the error return was moved to the fish function itself, rather than propagating out of the C++ (as far I understood it). Help is absolutely welcome.